suite/sanity/loss_recover.go

```diff
@@ -0,0 +1,173 @@
+package sanity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/gravitational/robotest/infra/gravity"
+	"github.com/gravitational/trace"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// K8S API master node
+	nodeApiMaster = "apim"
+	// Gravity Site master node
+	nodeGravitySiteMaster = "gsm"
+	// One of the GravitySite nodes
+	nodeGravitySiteNode = "gsn"
+	// Regular node
+	nodeRegularNode = "wrk"
+)
+
+type lossAndRecoveryParam struct {
+	// Timeouts is per-node operation timeout value
+	Timeouts gravity.OpTimeouts
+	// Role is node role
+	Role string
+	// InitialFlavor is equivalent to InitialNodes node
+	InitialFlavor string
+	// InitialNodes how many nodes
+	InitialNodes uint
+	// ReplaceNodes is how many nodes to loose and recover
+	ReplaceNodes uint
+	// ReplaceNodeType : see killXXX constants
+	ReplaceNodeType string
+	// ExpandBeforeShrink is whether to expand cluster before removing dead node
+	ExpandBeforeShrink bool
+	// PowerOff is whether to power off node before remove
+	PowerOff bool
+}
+
+func lossAndRecoveryVariety(template lossAndRecoveryParam) gravity.TestFunc {
+	var exp map[bool]string = map[bool]string{true: "expBfr", false: "expAft"}
```

Comment:
there's an 80 symbol limit to Azure resource group; if I'll go into long readable names, I'll have to start shortening them and that's where it'll become hardly readable in the console and log outputs for me.